clang/test/SemaCXX/bug141397.cpp

```diff
@@ -0,0 +1,16 @@
+// RUN: %clang_cc1 %s -fsyntax-only -verify
+// expected-no-diagnostics
+
+// This example uncovered a bug in Sema::BuiltinVectorMath, where we should be
+// using ASTContext::hasSameUnqualifiedType().
+
+typedef float vec3 __attribute__((ext_vector_type(3)));
+
+typedef struct {
+  vec3 b;
+} struc;
+
+vec3 foo(vec3 a,const struc* hi) {
+  vec3 b = __builtin_elementwise_max((vec3)(0.0f), a);
```

Comment:
I see four functions have been changed but I only see two different tests here. Can we please fully cover all the changes? If you feel that there are already tests covering the other changes, please point them out.

Reply:
It may be worth exploring further after you land this change. Thank you for following up and do more checking to verify.